docs/DesignDocs/Protected_Code_Loader_Proposal.md

```diff
@@ -0,0 +1,161 @@
+Open Enclave Protected Code Loader
+====
+
+Enabling Enclave Code Confidentiality in Open Enclave.
+
+# Motivation
+Current Open Enclave SDK provides code integrity and data confidentiality at
+run-time but not enclave binary confidentiality on a disk. The enclave binary
+is in plaintext and can be reverse engineered to reveal code logic and
+secret data embedded in the enclave image. 
+
+The Open Enclave Protected Code Loader shall enable providing confidentiality 
+and integrity to the IP sections in enclave image, i.e. , the ELF file.
+
+# User Experience
+## Enclave Build-time:
+### Linking pcl lib to the enclave 
+On building the enclave, a static lib "liboepcl.a" should be linked to the 
+enclave image with the compilation option "-Wl, whole-archive"
+### Encrypt the enclave image
+An encryption tool is provided as the pre-step to enclave signing process:
+
+```bash
+oeencrypt -i my_enclave -o my_enclave.enc -k keyfile
```

Comment:
@HernanGatta Can you comment on whether it would be feasible/desirable for OE SDK to provide common tooling for TA encryption as well on OP-TEE?